File: biliRecommendUI/src/main/java/com/bilirecommendui/utils/UIUtils.java
package com.bilirecommendui.utils;


import android.content.Context;
import android.view.inputmethod.InputMethodManager;
import android.widget.EditText;

import com.bilirecommendui.App;

/**
 * by y on 2016/9/13
 */
public class UIUtils {

    public static Context getContext() {
        return App.getContext();
    }

    public static String getSimpleName() {
        return getContext().getClass().getSimpleName();
    }


    public static float hypo(int a, int b) {
        return (float) Math.sqrt(Math.pow(a, 2) + Math.pow(b, 2));
    }


    public static void offKeyboard(EditText editText) {
        if (detectKeyboard(editText)) {
            InputMethodManager imm = (InputMethodManager) getContext().getSystemService(Context.INPUT_METHOD_SERVICE);
            imm.hideSoftInputFromWindow(editText.getWindowToken(), 0);
        }
    }

    public static void openKeyboard(EditText editText) {
        if (!detectKeyboard(editText)) {
            InputMethodManager imm = (InputMethodManager) getContext().getSystemService(Context.INPUT_METHOD_SERVICE);
            imm.showSoftInput(editText, InputMethodManager.SHOW_IMPLICIT);
        }
    }

    public static boolean detectKeyboard(EditText editText) {
        //true 弹出状态
        InputMethodManager imm = (InputMethodManager) getContext().getSystemService(Context.INPUT_METHOD_SERVICE);
        return imm.hideSoftInputFromWindow(editText.getWindowToken(), 0);
    }

}

File: BilibiliRecommendUi/app/src/main/java/com/bilibilirecommend/utils/UIUtils.kt
package com.bilibilirecommend.utils


import android.content.Context
import android.view.inputmethod.InputMethodManager
import android.widget.EditText

import com.bilibilirecommend.App


object UIUtils {

    val context: Context
        get() = App.instance

    val simpleName: String
        get() = context.javaClass.simpleName


    fun hypo(a: Int, b: Int): Float {
        return Math.sqrt(Math.pow(a.toDouble(), 2.0) + Math.pow(b.toDouble(), 2.0)).toFloat()
    }


    fun offKeyboard(editText: EditText) {
        if (detectKeyboard(editText)) {
            val imm = context.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
            imm.hideSoftInputFromWindow(editText.windowToken, 0)
        }
    }

    fun openKeyboard(editText: EditText) {
        if (!detectKeyboard(editText)) {
            val imm = context.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
            imm.showSoftInput(editText, InputMethodManager.SHOW_IMPLICIT)
        }
    }

    fun detectKeyboard(editText: EditText): Boolean {
        //true 弹出状态
        val imm = context.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
        return imm.hideSoftInputFromWindow(editText.windowToken, 0)
    }

}
